src/relayclient/RelayClient.ts

```diff
@@ -42,68 +40,72 @@ export interface RelayingResult {
-    this.web3 = web3
-    this.httpClient = httpClient
-    this.contractInteractor = contractInteractor
-    this.knownRelaysManager = knownRelaysManager
-    this.relayedTransactionValidator = relayedTransactionValidator
-    this.accountManager = accountManager
-    this.pingFilter = pingFilter
+    this.httpClient = dependencyTree.httpClient
+    this.contractInteractor = dependencyTree.contractInteractor
+    this.knownRelaysManager = dependencyTree.knownRelaysManager
+    this.transactionValidator = dependencyTree.transactionValidator
+    this.accountManager = dependencyTree.accountManager
+    this.pingFilter = dependencyTree.pingFilter
     this.relayHub = relayHub
-    this.asyncApprove = asyncApprove
+    this.asyncApprove = dependencyTree.asyncApprove
+  }
+
+  static getDefaultDependencies (provider: HttpProvider, gsnConfig: GSNConfig): TmpDependencyTree {
+    const httpWrapper = new HttpWrapper()
+    const httpClient = new HttpClient(httpWrapper, { verbose: false })
+    const accountManager = new AccountManager(provider, defaultEnvironment.chainId, gsnConfig.accountManagerConfig)
+    const contractInteractor = new ContractInteractor(provider, gsnConfig.contractInteractorConfig)
+    const knownRelaysManager = new KnownRelaysManager(gsnConfig.relayHubAddress, contractInteractor, EmptyFilter, gsnConfig.knownRelaysManagerConfig)
+    const transactionValidator = new RelayedTransactionValidator(contractInteractor, gsnConfig.relayHubAddress, defaultEnvironment.chainId, gsnConfig.transactionValidatorConfig)
+    return {
+      httpClient,
+      contractInteractor,
+      knownRelaysManager,
+      accountManager,
+      transactionValidator,
+      pingFilter: GasPricePingFilter,
+      asyncApprove: EmptyApprove
+    }
   }
 
   /**
    * Create an instance of {@link RelayClient} with all default implementations of its dependencies.
-   * @param web3
+   * @param provider
    * @param gsnConfig
    */
-  static new (web3: Web3, gsnConfig: GSNConfig): RelayClient {
-    const httpWrapper = new HttpWrapper()
-    const httpClient = new HttpClient(httpWrapper, { verbose: false })
-    const contractInteractor = new ContractInteractor(web3.currentProvider, gsnConfig.contractInteractorConfig)
-    const knownRelaysManager = new KnownRelaysManager(web3, gsnConfig.relayHubAddress, contractInteractor, EmptyFilter, gsnConfig.knownRelaysManagerConfig)
-    const accountManager = new AccountManager(web3, defaultEnvironment.chainId, gsnConfig.accountManagerConfig)
-    const transactionValidator = new RelayedTransactionValidator(contractInteractor, gsnConfig.relayHubAddress, defaultEnvironment.chainId, gsnConfig.transactionValidatorConfig)
-    return new RelayClient(web3, httpClient, contractInteractor, knownRelaysManager, accountManager, transactionValidator, gsnConfig.relayHubAddress, GasPricePingFilter, EmptyApprove, gsnConfig.relayClientConfig)
+  static new (provider: HttpProvider, gsnConfig: GSNConfig): RelayClient {
```

Comment:
don't depend on HttpProvider.

Reply:
What do I depend on then? I can define interface that has only 'send' method, cause it is the only method we use. Next PR, though.

Reply:
AFAIK i did saw you have `provider` type.